### ISSUE TITLE
Multiple variable declaration in the same line

### ISSUE THREAD

**Issue by ricardofbarros** [feross/standard#98]
Is this allowed?

```js
  var a, b, c

  a = 2
  b = 3
  c = 4

  a + 1
  b + 1
  c + 1
```
Standard isn't complaining about multiple variable declaration in the same line and in your examples you have variable declaration in each line.

**Comment by Flet** [comment]
Hmm, yeah not getting any warnings on this either.

```bash
flet@lappy:~/code$ cat blah.js 
var x, y, z
var q = 1, r = 2, s = 3
var a = 1, b, c
var m = 0
var n

a = 2
b = 3
c = 4

a + 1
b + 1

// semicolon to prove things are working!
c + 1;

x + y + z + q + r + s + a + b + c + m + n
flet@lappy:~/code$ standard blah.js 
Error: Use JavaScript Standard Style (https://github.com/feross/standard)
  blah.js:15:6: Extra semicolon.

flet@lappy:~/code$ standard --version
3.3.0

```

**Comment by Flet** [comment]
ah, looks like the shiny new `"one-var": [2, "never"],` is not turned on!

**Comment by Flet** [comment]
looks like turning this on breaks a lot of repos in test.js that are doing:
```js
var i, j, l, tmp, placeHolders, arr
```

**Comment by Flet** [comment]
https://github.com/eslint/eslint/issues/2206

**Comment by feross** [comment]
D'oh! Multiple variable declarations in a single line *should* only be allowed when they're all undefined, like:

```js
var a, b, c
```

I was so excited to release 3.3.0 to eliminate jscs and add the programmatic API that I forgot to enable `"one-var": [2, "never"]`. Let's wait until https://github.com/eslint/eslint/issues/2206 is resolved before enabling this rule.

**Comment by feross** [comment]
I put up $20 on issue https://github.com/eslint/eslint/issues/2206 in hopes it will get resolved soon. If anyone else wants to chip in, please do :+1:

**Comment by ricardofbarros** [comment]
Nice :smile:, should we close this issue since it's a issue related with eslint?

**Comment by feross** [comment]
No, let's leave it open so we don't forget.

**Comment by beaugunderson** [comment]
implemented in eslint/eslint@916277b2319d87fa6229d23fc045f46447ffab27 but not in a release yet, will be in >0.20.0

**Comment by feross** [comment]
Nice!

**Comment by feross** [comment]
We're still blocked on https://github.com/eslint/eslint/issues/2336. That needs to be fixed before we can start using the latest eslint.

**Comment by Flet** [comment]
Not confident it 2336 will be fixed soon... could turn down the padded blocks rule temporarily I suppose...

**Comment by Flet** [comment]
@feross any thoughts on relaxing the padded blocks rule in order to get to the latest eslint version?

**Comment by feross** [comment]
@Flet Yeah, let's just do that. The padded block rule isn't important for correctness or catching errors.

**Comment by feross** [comment]
I just tried adding the rule `"one-var": [2, { initialized: "never" }]`. Mostly works correctly, except it complains about:

```
var arr = [1, 2, 3];
for (var i = 0, len = arr.length; i < len; i++) {
  // ...
}
```

With this error: `Split initialized 'var' declarations into multiple statements.`

I'm inclined to hold off on this rule until we can ignore for loops somehow.

**Comment by feross** [comment]
Issue opened here: https://github.com/eslint/eslint/issues/2505

**Comment by feross** [comment]
ESLint finally fixed the bug where multiple vars were being checked in for statements!

Released 4.2.0 with the new rule: "Split initialized 'var' declarations into multiple statements". Arguably this could have been a semver major change, but not a single clone test failed (and we have nearly 80 repos in there!), so the risk is low.
